src/ucs/sys/sock.c

```diff
@@ -321,9 +328,13 @@ ucs_socket_handle_io_error(int fd, const char *name, ssize_t io_retval, int io_e
     }
 
     if (err_cb != NULL) {
-        status = err_cb(err_cb_arg, io_errno);
+        status = err_cb(err_cb_arg, io_errno, status);
         if (status == UCS_OK) {
-            return UCS_ERR_NO_PROGRESS;
+            /* UCS_ERR_CANCELED has to be returned if no other actions
+             * are required in order to prevent an endless loop in
+             * blocking IO operations (they continue a loop if UCS_OK
+             * or UCS_ERR_NO_PROGRESS is returned) */
+            return UCS_ERR_CANCELED;
         }
     }
 
```

Comment:
do we really need to print an error to the user in this case?
in case of many client-server connections, for example, if some peers disconnected as part of the app's flow, do we need to flood stdout for this? wdyt?
@yosefe @evgeny-leksikov